src/main/java/com/softserve/messaging/controller/MessageHandlerController.java

```diff
@@ -0,0 +1,32 @@
+package com.softserve.messaging.controller;
+
+import com.softserve.messaging.model.MessageHandler;
+import com.softserve.messaging.repository.MessageHandlerRepo;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/messaging/mailing")
```

Comment:
* your service is all about messaging, any reason to add a `messaging` prefix?
* it seems that your resource should be `Message` and endpoint `/api/messages`, `/api/messages/{id}`, `/api/messages?emailAddress=<email-address>`
* If your resource is generic e.g. "Message", then it shouldn't know about receipts. If not, your resource should have name "UserReceipt" or other to make clear that your class is not generic and knows about receipts